src/main/java/org/alfresco/repo/security/permissions/impl/acegi/ACLEntryAfterInvocationProvider.java

```diff
@@ -529,38 +529,19 @@ public void setPostProcessDenies(boolean postProcessDenies)
     
 	private ResultSet decide(Authentication authentication, Object object, ConfigAttributeDefinition config, ResultSet returnedObject) throws AccessDeniedException
     {
-    	ResultSet rs = optimisePermissionsCheck ? decideNew(authentication, object, config, returnedObject) :
-    			decideOld(authentication, object, config, returnedObject);
+        ResultSet rs = optimisePermissionsCheck ? decideNew(authentication, object, config, returnedObject) :
+                decideOld(authentication, object, config, returnedObject);
         return rs;
     }
- 
-    private ResultSet decideNew(Authentication authentication, Object object, ConfigAttributeDefinition config, ResultSet returnedObject) throws AccessDeniedException
 
+
+    private ResultSet decidePermissions(ResultSet returnedObject, List<ConfigAttributeDefintion> supportedDefinitions)

```

Comment:
Javadoc would be very helpful here to give some indication of what will happen in this large method.